kedro/framework/cli/cli.py

```diff
@@ -98,12 +100,13 @@ def docs():
 
 def _init_plugins():
     group = ENTRY_POINT_GROUPS["init"]
-    for entry_point in pkg_resources.iter_entry_points(group=group):
+    for entry_point in importlib_metadata.entry_points().select(group=group):
         try:
             init_hook = entry_point.load()
             init_hook()
-        except Exception as exc:
-            raise KedroCliError(f"Initializing {entry_point}") from exc
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.warning(KedroCliError(f"Fail to initialize {entry_point}"))
+            logger.warning(exc)
```

Comment:
```suggestion
            logger.warning(f"Failed to initialise %s. Full exception: %s", entry_point, exc)
```
Unless there's as good reason to use the `KedroCliError` still?